### PR TITLE
Hw 09

### PR DESCRIPTION
**Pull request by Resci**

### CODE REVIEW

**Review by tonitrus55 — APPROVED** [review]

File: src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java [hunk 1]
```diff
@@ -103,11 +108,35 @@ public boolean delete(Long id) {
     }
 
     private Driver parseFromResultSet(ResultSet resultSet) throws SQLException {
-        long driverId = resultSet.getObject("id", Long.class);
+        Long driverId = resultSet.getObject("id", Long.class);
+        if (driverId == null) {
```
Comment (Sofasmile) [inline src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java]: do we need this checking? Can we get null id from db?

File: src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java [hunk 2]
```diff
@@ -103,11 +108,35 @@ public boolean delete(Long id) {
...
         Driver driver = new Driver(name, licenseNumber);
+        driver.setLogin(login);
+        driver.setPassword(password);
         driver.setId(driverId);
         return driver;
     }
+
+    @Override
+    public Optional<Driver> findByLogin(String login) {
+        String selectQuery = "SELECT * FROM drivers where login LIKE ? and deleted = false";
```
Comment (Sofasmile) [inline src/main/java/com/mate/dao/jdbcimpl/DriverDaoJdbcImpl.java]: better don't use LIKE with login

File: src/main/java/com/mate/model/Driver.java [hunk 3]
```diff
@@ -6,12 +6,30 @@
     private Long id;
     private String name;
     private String licenceNumber;
+    private String login;
+    private String password;
```
Comment (Sofasmile) [inline src/main/java/com/mate/model/Driver.java:10]: we should add login and password to equals and hashcode

File: src/main/java/com/mate/security/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+import com.mate.exception.AuthenticationException;
+import com.mate.lib.Injector;
+import com.mate.lib.Service;
+import com.mate.model.Driver;
+import com.mate.service.DriverService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final Injector injector = Injector.getInstance("com.mate");
```
Comment (Sofasmile) [inline src/main/java/com/mate/security/AuthenticationServiceImpl.java]: do we need injector here? let's use Inject annotation instead

File: src/main/java/com/mate/security/AuthenticationServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    private final DriverService driverService = (DriverService) injector
+            .getInstance(DriverService.class);
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Driver driver = driverService.findByLogin(login).orElseThrow(
```
Comment (Sofasmile) [inline src/main/java/com/mate/security/AuthenticationServiceImpl.java]: Read about it in the Slack)

File: src/main/java/com/mate/web/filter/AuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+        String url = request.getServletPath();
+        if (url.equals("/drivers/add") || url.equals("/drivers/login")) {
```
Comment (Sofasmile) [inline src/main/java/com/mate/web/filter/AuthenticationFilter.java]: Read about it in the Slack)

File: src/main/java/com/mate/web/filter/AuthenticationFilter.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        Long driverId = (Long) request.getSession().getAttribute(DRIVER_ID);
+        if (driverService.get(driverId) == null) {
```
Comment (Sofasmile) [inline src/main/java/com/mate/web/filter/AuthenticationFilter.java]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/09_filters_checklist.html
- 5 paragraph

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/mate/web/filter/AuthenticationFilter.java [hunk 8]
```diff
@@ -30,13 +35,18 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
...
             return;
         }
 
         Long driverId = (Long) request.getSession().getAttribute(DRIVER_ID);
-        if (driverService.get(driverId) == null) {
+        try {
+            if (driverId == null) {
+                throw new NoSuchElementException("Value is null");
+            }
+            driverService.get(driverId);
```
Comment (Sofasmile) [inline src/main/java/com/mate/web/filter/AuthenticationFilter.java]: не варто тут відловлювати NoSuchElementException. Давай просто перевіряти чи driverId == null і робити:
response.sendRedirect("/drivers/login");
return;

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/mate/web/filter/AuthenticationFilter.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driver_id";
+    private static final Set<String> allowedUrls = new TreeSet<>();
+    private static final Injector injector = Injector.getInstance("com.mate");
+    private final DriverService driverService = (DriverService) injector
```
Comment (Sofasmile) [inline src/main/java/com/mate/web/filter/AuthenticationFilter.java]: do we need these fields?

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Comment by Sofasmile** [comment]
travis failed

**Comment by Resci** [comment]
> travis failed

Forget push commit

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**